IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m

```diff
@@ -0,0 +1,133 @@
+
+- (instancetype)initWithRequestParameters:(MSIDInteractiveRequestParameters *)parameters
+                             oauthFactory:(MSIDOauth2Factory *)oauthFactory
+                   tokenResponseValidator:(MSIDTokenResponseValidator *)tokenResponseValidator
+                               tokenCache:(id<MSIDCacheAccessor>)tokenCache
+                     accountMetadataCache:(MSIDAccountMetadataCacheAccessor *)accountMetadataCache
+{
+    self = [super initWithRequestParameters:parameters
+                               oauthFactory:oauthFactory
+                     tokenResponseValidator:tokenResponseValidator
+                                 tokenCache:tokenCache
+                       accountMetadataCache:accountMetadataCache];
+
+    if (self)
+    {
+        _extensionDelegate = [MSIDSSOExtensionTokenRequestDelegate new];
+        _extensionDelegate.context = parameters;
+        __weak typeof(self) weakSelf = self;
+        _extensionDelegate.completionBlock = ^(MSIDTokenResponse *response, NSError *error)
+        {
+            [weakSelf handleTokenResponse:response error:error completionBlock:weakSelf.requestCompletionBlock];
+        };
+        
+        _ssoProvider = [ASAuthorizationSingleSignOnProvider msidSharedProvider];
+    }
+
+    return self;
+}
+
+#pragma mark - MSIDInteractiveTokenRequest
+
+- (void)executeRequestWithCompletion:(MSIDInteractiveRequestCompletionBlock)completionBlock
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning interactive broker flow.");
+    
+    if (!completionBlock)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, self.requestParameters, @"Passed nil completionBlock. End silent broker flow.");
+        return;
+    }
+    
+    NSString *upn = self.requestParameters.accountIdentifier.displayableId ?: self.requestParameters.loginHint;
+    
+    [self.requestParameters.authority resolveAndValidate:self.requestParameters.validateAuthority
+                                           userPrincipalName:upn
+                                                     context:self.requestParameters
+                                             completionBlock:^(__unused NSURL *openIdConfigurationEndpoint,
+                                                               __unused BOOL validated, NSError *error)
+     {
+         if (error)
+         {
+             completionBlock(nil, error, nil);
+             return;
+         }
+        
+        NSError *localError;
+        __auto_type operationRequest = [MSIDBrokerOperationInteractiveTokenRequest tokenRequestWithParameters:self.requestParameters
+                                                                                                         error:&localError];
+        
+        if (!operationRequest)
+        {
+            completionBlock(nil, localError, nil);
+            return;
+        }
+        
+        ASAuthorizationSingleSignOnRequest *ssoRequest = [self.ssoProvider createRequest];
```

Comment:
nit: does this need to be executed from main thread?

Reply:
There is no such requirement.